### PR TITLE
[Bug fix] Fix the bug about map.keys and map.values

### PR DESCRIPTION
**Pull request by LiaoPeng**
1. Fix the about map.keys and map.values and set.values.
Because of initializing the array of keys and value with size, so using `push` will append data to array.

2. Add test cases for map.keys, map.values and set.values.

### CODE REVIEW

**Comment by dcodeIO** [comment]
Good find, seems this broke when I removed `Array.create`. LGTM, just a nit.

File: std/assembly/map.ts [hunk 1]
```diff
@@ -210,7 +210,7 @@ export class Map<K,V> {
     for (let i = 0; i < size; ++i) {
       let entry = changetype<MapEntry<K,V>>(start + <usize>i * ENTRY_SIZE<K,V>());
       if (!(entry.taggedNext & EMPTY)) {
-        keys.push(entry.key);
+        keys[length] = entry.key;
         ++length;
```
Comment (dcodeIO) [inline std/assembly/map.ts]: Can use `length++` here.

File: std/assembly/map.ts [hunk 2]
```diff
@@ -227,7 +227,7 @@ export class Map<K,V> {
     for (let i = 0; i < size; ++i) {
       let entry = changetype<MapEntry<K,V>>(start + <usize>i * ENTRY_SIZE<K,V>());
       if (!(entry.taggedNext & EMPTY)) {
-        values.push(entry.value);
+        values[length] = entry.value;
```
Comment (dcodeIO) [inline std/assembly/map.ts]: And here

**Comment by MaxGraey** [comment]
set.values has same problem: https://github.com/AssemblyScript/assemblyscript/blob/master/std/assembly/set.ts#L190. Could you fix that in this PR as well?

**Comment by LiaoPeng** [comment]
I'm glad to do this.

**Comment by dcodeIO** [comment]
Thanks!

**Pull request merged**